guidelines/enhancement_template.md

```diff
@@ -108,6 +108,42 @@ bogged down.
 
 Include a story on how this proposal will be operationalized:  lifecycled, monitored and remediated at scale.
```

Comment:
while you're in here, can we define what a good user story looks like?

it should:

1) clearly indicate who the user is   ("as a cluster administrator......"
2) what they want to achieve ("i want to see what permissions a user has")
3) why they want to achieve it ("so i can ensure no user has more permissions than i expect")

(it would also be good if we can disambiguate "Goals" and "user stories" but i don't have any great thoughts on how to explain that difference right now).

i think overall the EP template lacks "flow" in going from:

1) what is needed, who needs it, why they need it (user stories)
2) what that means we need to deliver (goals/non-goals)
3) how they will use this thing (workflow)
4) how we will build this thing (design/implementation)
5) how we will test/upgrade/support this thing
6) what concerns/risks/open questions exist around this thing

i find the "implementation details" and "Design details" sections particularly ambiguous (fundamentally i think the design details section is mistitled, given the subsections that fall under it)

Reply:
I've updated the instructions for user stories. 

I agree that some of the sections are poorly named or could be reordered, but that is probably best left to the next PR. I'll work on that after this merges.

Reply:
thanks, the user story instructions update looks good to me